### PR TITLE
feat(specs): revenue and filterEffects typing updates

### PR DESCRIPTION
**Pull request by cdhawke**
## 🧭 What and Why

🎟 JIRA Ticket: [OPTIM-748]

### Changes included:

- Fix `endAt` typing on `ABTest`
- Add Revenue typing for purchase, addToCart, and revenue significance, as well as metrics for `currencies` 
- Add `filterEffects` to display outlier data, taking the place of `outlier...Count` properties
- Add `configuration` on abtest creation `post` and retrieval `get` and `list`

## 🧪 Test

Green CI

[OPTIM-748]: https://algolia.atlassian.net/browse/OPTIM-748?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: specs/abtesting/common/schemas/ABTest.yml [hunk 1]
```diff
@@ -21,11 +21,35 @@ ABTest:
...
+        [A/B test significance](https://www.algolia.com/doc/guides/ab-testing/what-is-ab-testing/in-depth/how-ab-test-scores-are-calculated/#statistical-significance-or-chance) based on revenue data. A value of 0.95 or over is considered to be _significant_.
+      example:
+        USD: 1
+        EUR: 0.87
     updatedAt:
       $ref: '../parameters.yml#/updatedAt'
     createdAt:
       $ref: '../parameters.yml#/createdAt'
+    endAt:
+      $ref: '../parameters.yml#/endAt'
```
Comment (shortcuts) [inline specs/abtesting/common/schemas/ABTest.yml:52]: Just wondering, you said

> Fix endAt typing on ABTest

but I don't see a change to it in the PR, is this expected?

**Review by shortcuts — COMMENTED** [review]
looks great!

File: specs/abtesting/common/schemas/ABTest.yml (hunk 1 above)
Comment (cdhawke) [reply specs/abtesting/common/schemas/ABTest.yml:52]: The fix is [here](https://github.com/algolia/api-clients-automation/pull/2105/files#diff-ff712120fc90d0f77df3d9459902bf417378824a196ceb0a35c83b9d7906d34eL24) - `endAt` was previously nested under `conversionSignificance` where it should have been in its own `endAt` property:

```yml
conversionSignificance:
      type: number
      format: double
      description: >
        [A/B test significance](https://www.algolia.com/doc/guides/ab-testing/what-is-ab-testing/in-depth/how-ab-test-scores-are-calculated/#statistical-significance-or-chance) based on conversion. A value of 0.95 or over is considered to be _significant_.
      $ref: '../parameters.yml#/endAt'
```

Comment (shortcuts) [reply specs/abtesting/common/schemas/ABTest.yml:52]: ohhh goood catch, make sense!

**Review by JonathanMontane — COMMENTED** [review]
LGTM on the spec side as far as I can tell!

**Review by shortcuts — APPROVED** [review]
thanks!

**Pull request merged**